orchagent/neighorch.cpp

```diff
@@ -1240,7 +1294,15 @@ void NeighOrch::voqSyncAddNeigh(string &alias, IpAddress &ip_address, const MacA
     FieldValueTuple eiFv ("encap_index", to_string(attr.value.u32));
     attrs.push_back(eiFv);
 
-    FieldValueTuple macFv ("neigh", mac.to_string());
+    FieldValueTuple macFv;
+    if (inbandPort.m_type == Port::VLAN)
+    {
+        macFv = FieldValueTuple("neigh", gMacAddress.to_string());
```

Comment:
why we are putting our own mac when sync to chassis app db ?  remote side will use this to program SAI . Why we need to send our own mac to remote side ? as per design doc also SAI neighbor tables uses actual mac for vlan inband case. can you please clarify.

Reply:
We need to install kernel neighbor entries for remote neighbors. In inband vlan case, the mac of the kernel neighbor entry must be the router mac of remote asic, where the remote neighbor was learnt. This was described in section 2.5.2.2 of voq HLD.  Use neighbor nbr-0-1 as an example shown in that section. nbr-0-1 is learnt by asic0. A kernel neighbor entry is installed for the neighbor in asic1 with asic0's mac. This is why local router mac is published in chassis app db here.  SAI  programming  for remote neighbor uses encap index,  which is an indirection of the actual mac of remote neighbor.

Reply:
thanks. looks like  i overlooked it